### PR TITLE
[imgdata2hdf5] Tool to convert raw imgdata to hdf5

### PR DESCRIPTION
**Pull request by jinevening**
This adds imdata2hdf5, a tool to convert raw image data to hdf5

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to #3213 Draft PR #3232

### CODE REVIEW

**Comment by seanshpark** [comment]
as a random testing, I gave one text file as an item to list and got
```
Traceback (most recent call last):
  File "imgdata2hdf5.py", line 45, in <module>
    raw_data = bytearray(f.read())
TypeError: string argument without an encoding
```
and then one binary file as an item and got
```
Traceback (most recent call last):
  File "./imgdata2hdf5.py", line 45, in <module>
    raw_data = bytearray(f.read())
  File "/.../lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
```

File: compiler/imgdata2hdf5/imgdata2hdf5.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# Data list
+datalist = []
+with open(data_list, 'r') as f:
+    lines = f.readlines()
+    for line in lines:
+        filename = line.rstrip()
+        if os.path.isfile(filename):
+            datalist.append(filename)
+        else:
+            raise SystemExit("No such file. " + filename)
```
Comment (seanshpark) [inline compiler/imgdata2hdf5/imgdata2hdf5.py:42]: this will stop when there is an empty line or only with spaces that users can use as some delimiter

**Comment by seanshpark** [comment]
Still i get
```
Traceback (most recent call last):
  File "./imgdata2hdf5.py", line 52, in <module>
    raw_data = bytearray(f.read())
TypeError: string argument without an encoding
``` 
with some dummy text file as an input file

**Comment by seanshpark** [comment]
I used random text file as image file and a text list file with something like
```
/home/userid/some_random_textfile.txt

```
and an empty space at the bottom

**Comment by jinevening** [comment]
@seanshpark Can you upload your input text file? So I can solve your problem

**Comment by seanshpark** [comment]
> Can you upload your input text file? So I can solve your problem

Text file is random. you can use any file. You can event copy this text and use it as the input file.

File: compiler/imgdata2hdf5/imgdata2hdf5.py (hunk 1 above)
Comment (jinevening) [reply compiler/imgdata2hdf5/imgdata2hdf5.py:42]: I changed the code to support skipping of empty lines

**Comment by jinevening** [comment]
@seanshpark Sorry, I could not reproduce the below error

```
Traceback (most recent call last):
  File "./imgdata2hdf5.py", line 52, in <module>
    raw_data = bytearray(f.read())
TypeError: string argument without an encoding
```

I used random.txt, whose contents is

```
Text file is random. you can use any file. You can even copy this text and use it as the input file.
```

The output was 

```
No such file. Text file is random. you can use any file. You can even copy this text and use it as the input file
```

I also tried with data list file with 'random.txt', but it just ran well (and I think this is what we wanted).

Can you please upload your file? It can be easily attached in the comment.

**Comment by seanshpark** [comment]
I used your `Text file is random. you can use any file. You can even copy this text and use it as the input file.` file and got same error.

I'm not good at python but `TypeError: string argument without an encoding` can be a hint.

**Comment by seanshpark** [comment]
```
$ python --version
Python 3.5.2
```
and I'm running in TF2.3.0rc0 virtual environment

**Comment by jinevening** [comment]
> $ python --version
Python 3.5.2

This was the reason. I used Python 2.7, which does not generate the error. I've fixed the problem, so now the error has gone in both Python 3.X and Python 2.7.

**Pull request merged**